doc/source/whatsnew/v0.21.0.txt

```diff
@@ -172,6 +172,63 @@ We have updated our minimum supported versions of dependencies (:issue:`15206`,
    | Bottleneck   | 1.0.0           |          |
    +--------------+-----------------+----------+
 
+.. _whatsnew_0210.api_breaking.loc:
+
+Indexing with a list-like containing missing keys is Deprecated
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Previously, selecting at least 1 valid key with a list-like indexer would succeed and return ``NaN`` for non-found elements.
+This is exactly the function of ``.reindex()``. This will now show a ``FutureWarning`` message; in the future this will raise ``KeyError`` (:issue:`15747`).
+This warning will trigger on a ``DataFrame`` or a ``Series`` for using ``.loc[]`` on the Index, or ``[[]]`` on a ``Series``.
```

Comment:
I would remove `on the Index`, as you don't call `loc` on the index. 

Also, maybe we could just say that it is triggered for both `.loc[]` and `[]` (without needing to make the distinction of dataframe and series. Could mention after that that `[]` on dataframe already raises KeyError now)